app/controllers/publish/courses/modern_languages_controller.rb

```diff
@@ -64,59 +64,49 @@ def current_step
 
     private
 
-      def build_provider
-        @provider = RecruitmentCycle.find_by(year: params[:recruitment_cycle_year])
-                      .providers
-                      .find_by(provider_code: params[:provider_code])
+      def updated_subject_list
+        @updated_subject_list ||= selected_language_subjects_ids.concat(selected_non_language_subjects_ids)
+      end
+
+      def service
```

Comment:
Could we rename this to reflect the class name

Reply:
done